MESH-Research/CCR
# feat: improve authentication/registration user flow

Improve user login/register flow by:
- Using apolloclient state to store/fetch login status
- Router guards to protect routes that require login
- Redirect to dashboard after successful registration

## Review / Validation steps
1. `Lando rebuild -y`
2. Goto /
-   [x] "Register | login" should be in the top right
3. Click Register
4. Create an account
- [x] Should be redirected to `/dashboard`
- [x] "Username" with dropdown should be in topright nav
- [x] Refreshing browser should still show logged in in topright nav
- [x] Clicking Logout should redirect to `/` with "Register | Login" in navbar
5.  Login with username and password at `/login`
- [x] Redirect to /dashboard
6. Logout
7. Goto `/account/profile`
- [x] Should be redirected to `/login` with banner about needing to login
8. Login
- [x] Should be redirected to `/account/profile` rather than `/dashboard`
